### PR TITLE
[frontend] - add span links on synthetic requests

### PR DESCRIPTION
**Pull request by puckpuck**
Related to #329 

## Changes

- Creates a new trace if the inbound request has a `synthetic_request=true` baggage (from load generator)
- The new trace is linked to the inbound load generator trace using span links

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

Depends on #331

### CODE REVIEW

File: src/frontend/utils/telemetry/InstrumentationMiddleware.ts [hunk 1]
```diff
@@ -1,39 +1,51 @@
...
-    const { headers, method, url = '', httpVersion } = request;
-    const [target] = url.split('?');
+    return async (request, response) => {
+        const {headers, method, url = '', httpVersion} = request;
+        const [target] = url.split('?');
 
-    const parentContext = propagation.extract(context.active(), headers);
-    const span = await Tracer.createSpanFromContext(`API HTTP ${method}`, parentContext, { kind: SpanKind.SERVER });
+        let span;
+        const baggage = propagation.getBaggage(context.active());
```
Comment (xoscar) [inline src/frontend/utils/telemetry/InstrumentationMiddleware.ts:13]: nit: I think we can move most of this span creation logic inside of the BackendTracer functionality, we an just keep the span assignment

File: docs/trace_service_features.md [hunk 2]
```diff
@@ -11,10 +11,10 @@ Emoji Legend
...
-| Currency        | C++             | :no_bell:                 | :100:                | :100:                | :100:                   | :construction: | :100: |
+| Currency        | C++             | :no_bell:                 | :100:                | :100:                | :100:                   | :construction: | :100:          |
 | Email           | Ruby            | :100:                     | :100:                | :100:                | :construction:          | :construction: | :construction: |
 | Feature Flag    | Erlang / Elixir | :100:                     | :construction:       | :construction:       | :construction:          | :construction: | :construction: |
-| Frontend        | JavaScript      | :construction:            | :construction:       | :construction:       | :construction:          | :construction: | :construction: |
+| Frontend        | JavaScript      | :100:                     | :100:                | :100:                | :construction:          | :100:          | :100:          |
```
Comment (xoscar) [inline docs/trace_service_features.md:17]: 🔥

**Review by xoscar — APPROVED** [review]
Hey @puckpuck this is looking great! I left a small non-blocking comment

File: src/frontend/utils/telemetry/InstrumentationMiddleware.ts (hunk 1 above)
Comment (puckpuck) [reply src/frontend/utils/telemetry/InstrumentationMiddleware.ts:13]: If we moved this to the BackendTracer, we would still need to do all the attributes in the Middleware.

File: src/frontend/utils/telemetry/InstrumentationMiddleware.ts [hunk 3]
```diff
@@ -1,39 +1,51 @@
...
+        if (baggage?.getEntry("synthetic_request")?.value == "true") {
+            // if synthetic_request baggage is set, create a new trace linked to the span in context
+            // this span will look similar to the auto-instrumented HTTP span
+            const syntheticSpan = trace.getSpan(context.active()) as Span;
+            span = Tracer.getTracer().startSpan(`HTTP ${method}`, {
+                root: true,
+                kind: SpanKind.SERVER,
+                links: [{context: syntheticSpan.spanContext()}],
```
Comment (julianocosta89) [inline src/frontend/utils/telemetry/InstrumentationMiddleware.ts:21]: To be able to test this PR, we need to get #331 merged

**Review by julianocosta89 — APPROVED** [review]
It looks good to me, I'd just love to do a dry run before merging it

**Comment by puckpuck** [comment]
I put together this branch: https://github.com/honeycombio/opentelemetry-demo/tree/baggage-end-to-end which combines #331 and #332 

1. After starting it, go to Jaeger UI, and look for traces to the paymentservice (might take a minute before one appears)
2. This trace will have a `app.synthetic_requests=true` attribute on the frontend root span, and a `app.payment.charged=false` attribute on the paymentservice "charge" span. (also note the trace root has a link to the load generator trace).
3. Stop just the load generator.  From a new terminal window run `docker compose stop loadgenerator`
4. Using the app UI, place an order for a product
5. Return to Jaeger UI, and look for the trace you just created with the paymentservice
6. This trace will have a root span that comes from your browser `telemetry.sdk.language=webjs`.  There will be no attribute on this span or the subsequent frontend span about a synthetic request.
7. The paymentservice "charge" span will have `app.payment.charge=true` attribute for orders placed using a web browser.

**Comment by austinlparker** [comment]
@puckpuck can you resolve conflicts?

**Pull request merged**